docs/slide-in-use-cases.md

```diff
@@ -0,0 +1,64 @@
+# Overview through Slide In use cases
+
+It turned out that the sliding-in feature is more complex than it had appeared in the first place.
+This paper is an overview through the use cases.
+It should help to implement and understand the feature.
+
+## Slide In New Branch Dialog
+
+The picture below shows the **Slide In New Branch Dialog**.
+This is a regular **Create New Branch Dialog** with a changed title.
+
+![Slide In New Branch Dialog](slide-in-new-branch-dialog.png "Slide In New Branch Dialog")
+
+Note the checkbox options:
+- Checkout branch
+- Overwrite existing branch
+
+## Use cases
+
+| UC ID | Branch entry exists | Local branch exists | Estimated frequency |
+| :---: | :---: | :---: | :---: |
+| 1 | false | false | ~ 75% |
+| 2 | false | true | ~ 25% |
+| 3 | true | true | < 1% |
+| 4 | true | false | < 1% |
+
+## Slide In - Logic
+
+- When the **entry does not exist**, it must be added (slid in) under the given branch.
+- When the **entry exists under the given parent**, branch layout must not be affected.
+- When the **entry exists under other parent**, it must be first removed (slid out),
+and then added (slid in) under the given parent.
+
+## Slide In - New Branches
+
+This section applies to **UC1** and **UC4**.
+
+
+Use case **UC1** seems to be the most common.
+They are simply added in the way specified by the [slide in logic](#slide-in---logic).
+
+
+
+## Slide In - With "Overwrite existing branch"
+
+This section applies to **UC2** and **UC3**.
+
+
+**Overwriting a parent branch must not affect the branch layout**.
```

Comment:
Hmm why would we want to "overwrite the (commit pointed by?) parent branch"? If I add Y under X... why would X be overwritten in the process?

Reply:
Probably already answered below (added to doc too).

The overwrite option is required for existing branches. Without it the dialog's "Create" button is disabled.